### PR TITLE
added tests for fuse_bias, changed default usage

### PR DESCRIPTION
**Pull request by mstojkovicTT**
Added tests for fuse_bias, changed default from using fuse_bias to global param FORGE_FUSE_MATMUL_BIAS (was prevous FORGE_NO_FUSE_MATMUL_BIAS)

### CODE REVIEW

File: forge/csrc/forge_passes.cpp [hunk 1]
```diff
@@ -198,7 +198,9 @@ graphlib::Graph* run_pre_lowering_passes(
     recalculate_shapes(graph);
 
     // Fuse bias into matmuls
-    fuse_bias(graph);
+    if (env_as<bool>("FORGE_FUSE_MATMUL_BIAS")){
```
Comment (nvukobratTT) [inline forge/csrc/forge_passes.cpp:201]: It's okay to keep it disabled by default at the moment. However, let's create an issue with the label `tech dept` and link it here. In the long run, we would like to always fuse bias into matmul for added performance :))

File: forge/csrc/passes/pre_lowering_passes.cpp [hunk 2]
```diff
@@ -366,7 +363,7 @@ void fuse_bias(Graph *graph)
                 broadcast = true;
                 break;
             }
-
```
Comment (nvukobratTT) [inline forge/csrc/passes/pre_lowering_passes.cpp:369]: Make sure to run pre-commit checks as well :)) 

For more details, check this doc segment: https://docs.tenstorrent.com/tt-forge-fe/tools.html

File: forge/test/operators/matmul/test_matmul_fuse_bias.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import pytest
```
Comment (nvukobratTT) [inline forge/test/operators/matmul/test_matmul_fuse_bias.py:1]: General comment on import organization (sorted by hierarchy):
1. Native python libraries (e.g. os, math, etc.)
2. Third-party libraries (e.g. torch, numpy, etc.)
3. Our libraries/files (e.g. forge)

Grouping them like this will make our python files a bit more easier to read and follow :))

File: forge/test/operators/matmul/test_matmul_fuse_bias.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import torch
+from torch import nn
+
+import forge
+from forge.op.eval.common import compare_with_golden_pcc
+
+import os
+
+@pytest.mark.parametrize("dims", [
```
Comment (nvukobratTT) [inline forge/test/operators/matmul/test_matmul_fuse_bias.py:12]: These are more shapes as it seems :)) Also make sure to use formatted from pre-commit hooks :))

File: forge/test/operators/matmul/test_matmul_fuse_bias.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    (1, 784), (4, 784), (16, 784), (32, 784), (64, 784)
+])
+def test_matmul_bias_fusion(dims):
```
Comment (nvukobratTT) [inline forge/test/operators/matmul/test_matmul_fuse_bias.py:15]: Can you also add a test where matmul and bias are fused? 

This will be good to check if your logic works. Also, what you can explore, try to use compile depth from config.py file :)) Using it you'll be able to reduce the compile depth of fused matmul + bias test, and get passing results even if TTIR doesn't supports it :))

P.S.
You can also have single test, but use parameters to toggle bias True/False :))

File: forge/test/operators/matmul/test_matmul_fuse_bias.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import torch
+from torch import nn
+
+import forge
+from forge.op.eval.common import compare_with_golden_pcc
+
+import os
+
+@pytest.mark.parametrize("dims", [
+    (1, 784), (4, 784), (16, 784), (32, 784), (64, 784)
```
Comment (nvukobratTT) [inline forge/test/operators/matmul/test_matmul_fuse_bias.py:13]: One more advice, as 784 dim is appearing in every case, you can make it as separate pytest param, and only iterate over batch param :))

**Comment by nvukobratTT** [comment]
@mstojkovicTT one general comment, please rebase your changes on top of the latest main, so you can get CI results :))

image

File: forge/test/operators/matmul/test_matmul_fuse_bias.py (hunk 3 above)
Comment (mstojkovicTT) [reply forge/test/operators/matmul/test_matmul_fuse_bias.py:4]: fixed

File: forge/test/operators/matmul/test_matmul_fuse_bias.py (hunk 4 above)
Comment (mstojkovicTT) [reply forge/test/operators/matmul/test_matmul_fuse_bias.py]: fixed

File: forge/test/operators/matmul/test_matmul_fuse_bias.py (hunk 6 above)
Comment (mstojkovicTT) [reply forge/test/operators/matmul/test_matmul_fuse_bias.py]: done

File: forge/test/operators/matmul/test_matmul_fuse_bias.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mstojkovicTT) [reply forge/test/operators/matmul/test_matmul_fuse_bias.py]: done

File: forge/csrc/forge_passes.cpp (hunk 1 above)
Comment (mstojkovicTT) [reply forge/csrc/forge_passes.cpp]: done!

**Review by pilkicTT — APPROVED** [review]

**Review by nvukobratTT — APPROVED** [review]
